### ISSUE TITLE
[Bug]: `--endpoint` parameter does not accept fully qualified name

### ISSUE THREAD

**Issue by gmatev** [alcionai/corso#2330]
### What happened?

```
corso repo init s3 --endpoint=http://127.0.0.1:9000 --bucket foobar
Error: Failed to initialize a new S3 repository: initializing repo: unable to create client: Endpoint url cannot have fully qualified paths.
```

The behavior is unintuitive especially for a user trying to connect to an HTTP endpoint for testing. 

One of the following should happen:
* Option 1:
   * Fix inline help and error message to indicate that `http://` vs `https://` is controlled via the `--disable-tls` flag  
   * Should provide a specific example. Even the example today uses a fully qualified path including the protocol for the endpoint
* Option 2:
   *  Optionally accept `http:// | https://`. If not specified, current behavior (default to `https://` unless `disable-tls` is specified) applies
   * If prefix is `https://` make sure is is not specified in conjunction with `disable-tls` since the two are incompatible
   * If prefix is `http://`, `--disable-tls` is implied but can still be specified
* 

### Corso Version?

Corso v0.2.0

### Where are you running Corso?

OS: MacOS 12.5
Storage: Minio

**Comment by ntolia** [comment]
@gmatev This behavior is inherited from kopia choices and we should just keep those options as-is. Today, by default, we try TLS. We can allow for both insecure TLS (`https` w/o cert validation) and no TLS via flags (`http`). This behavior is consistent across the flags and I don't know if adding a translation layer is worth it at this moment.

**Comment by gmatev** [comment]
We should at least enhance the inline help for the flags to clarify (Option 1). We had multiple people trip on this until we got it right. 

Also someone :) once sent me a Kopia example to use with Minio which did include `--endpoint http://127.0.0.1:9000` so unfortunately it tends to be the natural inclination. Clarifying the error message and the help parameters should help.

**Comment by ntolia** [comment]
@gmatev Yes, I agree with option 1 here. Also, I really wonder who that someone was 🤔.

**Comment by neha-Gupta1** [comment]
`repo connect` command seems to have similar issue. Should I update the error message there too?
